packages/driver/cypress/integration/e2e/multi-domain/multi_domain_spec.ts

```diff
@@ -278,4 +287,223 @@ describe('multi-domain', { experimentalSessionSupport: true, experimentalMultiDo
 
     it('short circuits the secondary domain command queue when "done()" is called early')
   })
+
+  describe('yields', () => {
+    it('yields a value', () => {
+      cy.switchToDomain('foobar.com', () => {
+        cy
+        .get('[data-cy="dom-check"]')
+        .invoke('text')
+      }).should('equal', 'From a secondary domain')
+    })
+
+    it('yields the cy value even if a return is present', () => {
+      cy.switchToDomain('foobar.com', async () => {
+        cy
+        .get('[data-cy="dom-check"]')
+        .invoke('text')
+
+        const p = new Promise((resolve, reject) => {
+          setTimeout(() => {
+            resolve('text')
+          }, 1000)
+        })
+
+        return p
+      }).should('equal', 'From a secondary domain')
+    })
+
+    it('errors if a cy command is present and it returns a sync value', (done) => {
+      cy.on('fail', (err) => {
+        assertLogLength(logs, 6)
+        expect(logs[5].get('error')).to.eq(err)
+        expect(err.message).to.include('`cy.switchToDomain()` failed because you are mixing up async and sync code.')
+
+        done()
+      })
+
+      cy.switchToDomain('foobar.com', () => {
+        cy
+        .get('[data-cy="dom-check"]')
+        .invoke('text')
+
+        return 'text'
+      })
+    })
+
+    it('yields synchronously', () => {
+      cy.switchToDomain('foobar.com', () => {
+        return 'From a secondary domain'
+      }).should('equal', 'From a secondary domain')
+    })
+
+    it('yields asynchronously', () => {
+      cy.switchToDomain('foobar.com', async () => {
+        return new Promise((resolve: (val: string) => any, reject) => {
+          setTimeout(() => {
+            resolve('From a secondary domain')
+          }, 1000)
```

Comment:
Can this be shortened to something like 50-100ms?

Reply:
it sure can!